File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    var answer = [];
  
        for(let i = 0; i < arr.length;){
            let lastIndex = answer.length - 1;
            
            if(answer.length == 0){
                answer.push(arr[i++]);
            }
            else if(answer[lastIndex] == arr[i]){
                answer.splice(lastIndex, 1);
                i++;
            }
            else{
                answer.push(arr[i++]);
            }
        }
        
        if(answer.length == 0){
            let t = [];
            t.push(-1);
            return t;
        };
        return answer;
    return answer;
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] arr) {
        List<int> answer = new List<int>();
        for(int i = 0; i < arr.Length;){
            int lastIndex = answer.Count - 1;
            
            if(answer.Count == 0){
                answer.Add(arr[i++]);
            }
            else if(answer[lastIndex] == arr[i]){
                answer.RemoveAt(lastIndex);
                i++;
            }
            else{
                answer.Add(arr[i++]);
            }
        }
        
        if(answer.Count == 0) return new int[]{-1};
        return answer.ToArray();
    }
}